src/common.jl

```diff
@@ -1,12 +1,14 @@
 ## Generic ##
 
+Base.one(::Irrational) = 1
```

Comment:
Zygote fails with any distribution that involves an irrational number because of this. This started happening in Distributions 0.22.3 but did not happen in 0.22.2. I can define an adjoint here instead of actually defining a method to make this less extreme.

Reply:
So this is a bit harder than I thought. Here is a MWE that fails without defining `one` on `Irrational`:
```julia
Zygote.gradient(x -> x[1] + π, [1.0])
```
A lot of binary functions in `DiffRules` assume that `one` is defined from both arguments.

Reply:
https://github.com/JuliaLang/julia/pull/32117

Reply:
Given that `one(Real)` is `1`, I think it kind of makes sense that `one(::Irrational)` is also `1` by default for lack of a better alternative. `true` is another option.

Reply:
> true is another option

The advantage of `true` would be that it promotes less.

BTW I guess the main problem is actually DiffRules 1.0.1 (https://github.com/JuliaDiff/DiffRules.jl/compare/v1.0.0...v1.0.1) which is introduced all `one` based definitions. The only difference between Distributions 0.22.2 and 0.22.3 that could cause problems seems to be that the newer version supports SpecialFunctions 0.10?

Reply:
You are probably right. The errors do trace back to `DiffRules`.

Reply:
https://github.com/JuliaLang/julia/pull/34773

Reply:
This seems to be an acceptable solution according to Slack and the PR above and it will likely be there in future Julia versions. Whether we should define it here or in Compat for the current Julia versions is not a show stopper imo, so we can merge now and fix later.